frontend/src/metabase-types/api/settings.ts

```diff
@@ -352,4 +353,7 @@ export type Settings = InstanceSettings &
 
 export type SettingKey = keyof Settings;
 
-export type SettingValue = Settings[SettingKey];
+export type SettingValue<Key extends SettingKey = SettingKey> =
```

Comment:
Why do we need this?
AFAIK settings have an initial state that's coming from a server rendered thing, so they should never not have a value

Reply:
all good now, it was needed for `SettingDefinition`